zerver/middleware.py

```diff
@@ -307,6 +308,7 @@ class RequestContext(MiddlewareMixin):
     def __call__(self, request: HttpRequest) -> HttpResponse:
         set_request(request)
         try:
+            assert self.get_response is not None
```

Comment:
`get_response=None` is deprecated as of 3.1a1 (django/django@4d973f593932285cd2f765400d915305d8e7333a) and impossible as of 4.0a1(django/django@4bb30fe5d598a7acd2a3055c5e66224cf42a75e9). Perhaps this should be adjusted in django-stubs.

Reply:
I see. Will we one day replace `MiddlewareMixin` with the newer `__call__` API one day?

Reply:
Unclear. At some point we’ll want to update all our middleware with async support (so we can finally destroy our `AsynchronousResponse` hack). `MiddlewareMixin` conveniently does that automatically, though not in the most optimal way.